### PR TITLE
Scripting: Remove native scripts

### LINKED ISSUE #19966: Remove native scripts and replace with script engine example
Native scripts are heavyweight things that require creating a plugin. They are used by advanced users who need access to special code (eg a custom library to do something not available in the exposed methods/classes in painless or groovy).  Internally, we have a single NativeScriptEngine which handles taking the registered native scripts, and making them accessible. But writing a NativeScriptFactory isn't much different than writing a ScriptEngine.

I think we should simplify scripting internals by removing NativeScriptEngine, and documenting instead how to create a ScriptEngine for this advanced case.

### PR DESCRIPTION
**Pull request by rjernst**
Native scripts have been replaced in documentation by implementing
a ScriptEngine and they were deprecated in 5.5.0. This commit
removes the native script infrastructure for 6.0.

closes #19966

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/script/ScriptFieldIT.java [hunk 1]
```diff
@@ -1,184 +0,0 @@
...
-    protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Arrays.asList(CustomScriptPlugin.class);
-    }
-
-    static int[] intArray = { Integer.MAX_VALUE, Integer.MIN_VALUE, 3 };
-    static long[] longArray = { Long.MAX_VALUE, Long.MIN_VALUE, 9223372036854775807L };
-    static float[] floatArray = { Float.MAX_VALUE, Float.MIN_VALUE, 3.3f };
-    static double[] doubleArray = { Double.MAX_VALUE, Double.MIN_VALUE, 3.3d };
-
-    public void testNativeScript() throws InterruptedException, ExecutionException {
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/script/ScriptFieldIT.java]: I believe we have plenty of tests for script fields hanging around though there may not be any in core if you drop this. Is that a problem, do you think?

**Review by nik9000 — APPROVED** [review]
I like all the removed lines! I'm fine with pushing folks over to a script engine. @imotov probably should have a look as well if just so that he knows this is happening.

File: core/src/test/java/org/elasticsearch/script/ScriptFieldIT.java (hunk 1 above)
Comment (rjernst) [inline core/src/test/java/org/elasticsearch/script/ScriptFieldIT.java]: This wasn't really testing script fields per say, but I will double check there is some test for script fields.

**Review by imotov — APPROVED** [review]
LGTM

**Pull request merged**